src/Libs/cairo-1.0/Records/Surface.cs

```diff
@@ -0,0 +1,68 @@
+﻿namespace Cairo
+{
+    public partial class Surface
+    {
+        public Content Content => Internal.Surface.GetContent(Handle);
+        public Device Device => new Device(Internal.Surface.GetDevice(Handle));
+        public Status Status => Internal.Surface.Status(Handle);
+        public SurfaceType SurfaceType => Internal.Surface.GetType(Handle);
+
+        public (double X, double Y) DeviceOffset
+        {
+            get
+            {
+                Internal.Surface.GetDeviceOffset(Handle, out double x_offset, out double y_offset);
+                return (x_offset, y_offset);
+            }
+            set => Internal.Surface.SetDeviceOffset(Handle, value.X, value.Y);
+        }
+
+        public (double X, double Y) DeviceScale
+        {
+            get
+            {
+                Internal.Surface.GetDeviceScale(Handle, out double x_scale, out double y_scale);
+                return (x_scale, y_scale);
+            }
+            set => Internal.Surface.SetDeviceScale(Handle, value.X, value.Y);
+        }
+
+        public (double X, double Y) FallbackResolution
+        {
+            get
+            {
+                Internal.Surface.GetFallbackResolution(Handle, out double x_pixels_per_inch, out double y_pixels_per_inch);
+                return (x_pixels_per_inch, y_pixels_per_inch);
+            }
+            set => Internal.Surface.SetFallbackResolution(Handle, value.X, value.Y);
+        }
+
+        public FontOptions FontOptions
```

Comment:
The font options can be mutable (e.g. `cairo_font_options_set_antialias()`) so I think it's better not to cache anything.
Changing it to `FontOptions GetFontOptions()` sounds okay to me, or it could be `void GetFontOptions(FontOptions opts)` if we want to let a user fill in their own font options instance. Are there any conventions for that in the generated code when there's a single "out" parameter in the C API?

Reply:
No. The api is generated like defined in the gir-file (I'm not sure if there is support for out parameter at all currently).

I think you are right and we should just mirror the C api and make it an out parameter,too.